core/store/src/trie/mod.rs

```diff
@@ -734,6 +743,81 @@ impl Trie {
     pub fn get_trie_nodes_count(&self) -> TrieNodesCount {
         self.storage.get_trie_nodes_count()
     }
+
+    /// Given a root and a key, it returns a MerkleProof for that key.
+    /// Note that it can be a proof of membership, or a proof of non membership. This is encoded
+    /// in the boolean of the return type.
+    ///
+    /// Hence, if [get_proof](#method.get_proof) returns (true, _), it means that the key-value are found in the trie
+    /// and a proof for that is presented. Similarly, if (false, _) is returned, it means that the key-value
+    /// pair does not belong to the trie and a proof for that is also returned.
+    ///
+    /// # Errors
+    ///
+    /// If the root or any subsequent is not found in the trie, `StorageError` will be returned
+    pub fn get_proof(
+        &self,
+        root: &CryptoHash,
+        key: &[u8],
+    ) -> Result<(bool, Vec<RawTrieNodeWithSize>), StorageError> {
+        let mut key = NibbleSlice::new(key);
+        let mut hash = *root;
+        if hash == Trie::EMPTY_ROOT {
+            return Ok((key.is_empty(), Vec::new()));
+        }
+
+        let mut levels: Vec<RawTrieNodeWithSize> = vec![];
+        loop {
+            let bytes = self.storage.retrieve_raw_bytes(&hash)?;
+            let node = RawTrieNodeWithSize::decode(&bytes).map_err(|_| {
+                StorageError::StorageInconsistentState("RawTrieNode decode failed".to_string())
```

Comment:
```suggestion
            let node = RawTrieNodeWithSize::decode(&bytes).map_err(|err| {
                StorageError::StorageInconsistentState(format!("RawTrieNode decode failed with {}", err))
```

Why not include the underlying error?